src/test/java/com/google/sps/servlets/RentPostsServletTest.java

```diff
@@ -0,0 +1,160 @@
+package com.google.sps;
+
+import static com.google.sps.Constants.PROJECT_ID;
+import static com.google.sps.Constants.RENT_COLLECTION_NAME;
+import static com.google.sps.Constants.REQUEST_DESCRIPTION;
+import static com.google.sps.Constants.REQUEST_END_DATE;
+import static com.google.sps.Constants.REQUEST_LEASE_TYPE;
+import static com.google.sps.Constants.REQUEST_NUM_ROOMS;
+import static com.google.sps.Constants.REQUEST_PRICE;
+import static com.google.sps.Constants.REQUEST_ROOM_TYPE;
+import static com.google.sps.Constants.REQUEST_START_DATE;
+import static com.google.sps.Constants.REQUEST_TIMESTAMP;
+import static com.google.sps.Constants.REQUEST_TITLE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.*;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions.Builder;
+import com.google.cloud.firestore.Query;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.WriteResult;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class RentPostsServletTest {
+  @Mock CollectionReference collectionMock;
+  @Mock Firestore dbMock;
+  @Mock HttpServletRequest request;
+  @Mock HttpServletResponse response;
+  private RentPostsServlet rentPostsServlet;
+
+  @Before
+  public void setUp() throws Exception {    
+    MockitoAnnotations.initMocks(this);
+    rentPostsServlet = new RentPostsServlet();
+    rentPostsServlet.init();
+    FirestoreUtil.setDatabase(dbMock);
+  }
+
+  @Test
+  public void testDatabaseMock_MockExistsAndIsUsedByServlet() throws Exception {
+    assertNotNull(FirestoreUtil.getDatabase());
+    assertEquals(dbMock, FirestoreUtil.getDatabase());
+  }
```

Comment:
This seems unnecessary. If these are necessary for your actual tests would fail without them. If they're not necessary for your tests, then there's no reason to be testing them anyway.

Reply:
Removed the test.